.changeset/large-melons-warn.md

```diff
@@ -0,0 +1,5 @@
+---
+'slate-react': minor
+---
+
+Fix CJK IME(like Chinese, Japanese) double input
```

Comment:
```suggestion
Fix CJK IME (e.g. Chinese or Japanese) double input
```

Reply:
Now the purpose of processing in beforeinput is for the processing of WeChat browser, So just use it for onCompositionEnd processing, the two are duplicates
```
if (type === 'insertFromComposition') {
  //.....
}
```
insertFromComposition is to adapt to the second level of input that has been implemented, so there should not be redundant processing, and user agent should not be used to judge the browser, this is not a long-term consideration.

Reply:
I contacted ttitoo and confirmed this issue